File: Easy/upb/meta_upbdefaultview/meta_upbdefaultview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_upbdefaultview
{
    public class Meta_upbdefaultview : Meta_easydata 
	{
        public Meta_upbdefaultview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "upbdefaultview") {
				Name = "Unità previsionali di base";
			EditTypes.Add("default");
			ListingTypes.Add("default");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idupb"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "default": {
						DescribeAColumn(T, "idupb", "id voce upb (tabella upb)", nPos++);
						DescribeAColumn(T, "title", "Denominazione", nPos++);
						DescribeAColumn(T, "upb_active", "attivo", nPos++);
						DescribeAColumn(T, "upb_assured", "Finanziamento certo (Non gestire assegnazione crediti/incassi)", nPos++);
						DescribeAColumn(T, "upb_cigcode", "Codice CIG, Codice identificativo di gara", nPos++);
						DescribeAColumn(T, "upb_codeupb", "codice upb", nPos++);
						DescribeAColumn(T, "upb_cofogmpcode", "Cofogmpcode", nPos++);
						DescribeAColumn(T, "upb_cupcode", "Codice CUP, Codice unico di progetto", nPos++);
						DescribeAColumn(T, "upb_expiration", "scadenza", nPos++);
						DescribeAColumn(T, "upb_flag", "flag vari", nPos++);
						DescribeAColumn(T, "upb_flagactivity", "Tipo attività", nPos++);
						DescribeAColumn(T, "upb_flagkind", "Funzione", nPos++);
						DescribeAColumn(T, "upb_granted", "Finanziamento concesso", nPos++);
						if (T.Columns.Contains("upb_granted")) T.Columns["upb_granted"].ExtendedProperties["format"] = "fixed.2";
						DescribeAColumn(T, "epupbkind_title", "Denominazione ID Tipo UPB nell'economico patrimoniale (tabella epupbkind)", nPos++);
						DescribeAColumn(T, "epupbkind_description", "Descrizione ID Tipo UPB nell'economico patrimoniale (tabella epupbkind)", nPos++);
						DescribeAColumn(T, "treasurer_description", "Id cassiere (tabella treasurer)", nPos++);
						DescribeAColumn(T, "underwriter_description", "ID Ente finanziatore (tabella underwriter)", nPos++);
						DescribeAColumn(T, "upb_newcodeupb", "Codice di consolidamento", nPos++);
						DescribeAColumn(T, "upbparent_title", "chiave parent U.P.B. (tabella upb) ", nPos++);
						DescribeAColumn(T, "upb_previousappropriation", "Totale impegnato pregresso (previa informatizzazione)", nPos++);
						if (T.Columns.Contains("upb_previousappropriation")) T.Columns["upb_previousappropriation"].ExtendedProperties["format"] = "fixed.2";
						DescribeAColumn(T, "upb_previousassessment", "Totale accertato pregresso (previa informatizzazione)", nPos++);
						if (T.Columns.Contains("upb_previousassessment")) T.Columns["upb_previousassessment"].ExtendedProperties["format"] = "fixed.2";
						DescribeAColumn(T, "upb_printingorder", "Ordine di stampa", nPos++);
						DescribeAColumn(T, "upb_requested", "Finanziamento richiesto", nPos++);
						if (T.Columns.Contains("upb_requested")) T.Columns["upb_requested"].ExtendedProperties["format"] = "fixed.2";
						DescribeAColumn(T, "upb_rtf", "allegati", nPos++);
						DescribeAColumn(T, "upb_ri_ra_quota", "Ri_ra_quota", nPos++);
						if (T.Columns.Contains("upb_ri_ra_quota")) T.Columns["upb_ri_ra_quota"].ExtendedProperties["format"] = "fixed.6";
						DescribeAColumn(T, "upb_start", "data inizio", nPos++);
						DescribeAColumn(T, "upb_ri_rb_quota", "Ri_rb_quota", nPos++);
						if (T.Columns.Contains("upb_ri_rb_quota")) T.Columns["upb_ri_rb_quota"].ExtendedProperties["format"] = "fixed.6";
						DescribeAColumn(T, "upb_stop", "data fine", nPos++);
						DescribeAColumn(T, "upb_ri_sa_quota", "Ri_sa_quota", nPos++);
						if (T.Columns.Contains("upb_ri_sa_quota")) T.Columns["upb_ri_sa_quota"].ExtendedProperties["format"] = "fixed.6";
						DescribeAColumn(T, "upb_txt", "note testuali", nPos++);
						DescribeAColumn(T, "upb_uesiopecode", "Uesiopecode", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "default": {
						return "title asc ";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/upb/meta_upbdefaultview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_upbdefaultview() {
        MetaData.apply(this, ["upbdefaultview"]);
        this.name = 'meta_upbdefaultview';
    }

    meta_upbdefaultview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_upbdefaultview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'default':
						this.describeAColumn(table, 'idupb', 'id voce upb (tabella upb)', null, 10, 36);
						this.describeAColumn(table, 'title', 'Denominazione', null, 20, 150);
						this.describeAColumn(table, 'upb_active', 'Attivo', null, 30, null);
						this.describeAColumn(table, 'upb_assured', 'Finanziamento certo (Non gestire assegnazione crediti/incassi)', null, 40, null);
						this.describeAColumn(table, 'upb_cigcode', 'Codice CIG, Codice identificativo di gara', null, 50, 10);
						this.describeAColumn(table, 'upb_codeupb', 'Codice', null, 60, 50);
						this.describeAColumn(table, 'upb_cupcode', 'Codice CUP, Codice unico di progetto', null, 70, 15);
						this.describeAColumn(table, 'upb_expiration', 'scadenza', null, 80, null);
						this.describeAColumn(table, 'upb_flag', 'flag vari', null, 90, null);
						this.describeAColumn(table, 'upb_flagactivity', 'Tipo attività', null, 100, null);
						this.describeAColumn(table, 'upb_flagkind', 'Funzione', null, 110, null);
						this.describeAColumn(table, 'upb_granted', 'Finanziamento concesso', 'fixed.2', 120, null);
						this.describeAColumn(table, 'epupbkind_title', 'Denominazione ID Tipo UPB nell\'economico patrimoniale (tabella epupbkind)', null, 130, 50);
						this.describeAColumn(table, 'epupbkind_description', 'Descrizione ID Tipo UPB nell\'economico patrimoniale (tabella epupbkind)', null, 130, 500);
						this.describeAColumn(table, 'treasurer_description', 'Id cassiere (tabella treasurer)', null, 200, 150);
						this.describeAColumn(table, 'underwriter_description', 'ID Ente finanziatore (tabella underwriter)', null, 210, 50);
						this.describeAColumn(table, 'upb_newcodeupb', 'Codice di consolidamento', null, 220, 50);
						this.describeAColumn(table, 'upbparent_title', 'chiave parent U.P.B. (tabella upb) ', null, 230, 150);
						this.describeAColumn(table, 'upb_previousappropriation', 'Totale impegnato pregresso (previa informatizzazione)', 'fixed.2', 240, null);
						this.describeAColumn(table, 'upb_previousassessment', 'Totale accertato pregresso (previa informatizzazione)', 'fixed.2', 250, null);
						this.describeAColumn(table, 'upb_printingorder', 'Ordine di stampa', null, 260, 50);
						this.describeAColumn(table, 'upb_requested', 'Finanziamento richiesto', 'fixed.2', 270, null);
						this.describeAColumn(table, 'upb_rtf', 'allegati', null, 280, null);
						this.describeAColumn(table, 'upb_start', 'data inizio', null, 290, null);
						this.describeAColumn(table, 'upb_stop', 'data fine', null, 300, null);
						this.describeAColumn(table, 'upb_txt', 'note testuali', null, 310, null);
						this.describeAColumn(table, 'upb_cofogmpcode', 'Cofogmpcode', null, 400, 10);
						this.describeAColumn(table, 'upbcapofila_title', 'Idupb_capofila', null, 410, 150);
						this.describeAColumn(table, 'upb_ri_ra_quota', 'Ri_ra_quota', 'fixed.6', 420, null);
						this.describeAColumn(table, 'upb_ri_rb_quota', 'Ri_rb_quota', 'fixed.6', 430, null);
						this.describeAColumn(table, 'upb_ri_sa_quota', 'Ri_sa_quota', 'fixed.6', 440, null);
						this.describeAColumn(table, 'upb_uesiopecode', 'Uesiopecode', null, 450, 10);
//$objCalcFieldConfig_default$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idupb"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "default": {
						return "title asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('upbdefaultview', new meta_upbdefaultview('upbdefaultview'));

	}());
